ValveResourceFormat/IO/FileExtract.cs

```diff
@@ -1,16 +1,39 @@
 using System;
+using System.Collections.Generic;
 using System.IO;
 using System.Text;
 using SkiaSharp;
 using ValveResourceFormat.ResourceTypes;
 
 namespace ValveResourceFormat.IO
 {
+
+    public class ExtractedResource
+    {
+        public byte[] Data { get; set; }
+        public List<ChildExtractedResource> Children { get; private set;}
+
+        public ExtractedResource()
+        {
+            Children = new List<ChildExtractedResource>();
+        }
+    }
+
+    public class ChildExtractedResource
+    {
+        public byte[] Data { get; set; }
```

Comment:
Aren't you going to consume a lot of memory trying to keep all the child resources in memory instead of writing them to disk as they come?

Reply:
Also am I blind, where does ChildExtractedResource even get created?

Reply:
Example usage: https://github.com/kristiker/ValveResourceFormat/blob/f7d417652db94b5cc5a048a1f5da45a970d99e63/ValveResourceFormat/IO/FileExtract.cs#L93